wagtail_streamfield_migration_toolkit/test/tests/test_migrations.py

```diff
@@ -0,0 +1,353 @@
+                new_content = json.loads(new_revision.content["content"])
+                for old_block, new_block in zip(old_content, new_content):
+                    self.assertEqual(old_block["id"], new_block["id"])
+                    if is_latest_or_live and old_block["type"] == "char1":
+                        self.assertEqual(new_block["type"], "renamed1")
+                    else:
+                        self.assertEqual(old_block["type"], new_block["type"])
+
+    def test_migrate_revisions_from_date(self):
+        revisions_from = datetime.datetime.now() - datetime.timedelta(days=2)
+        self.apply_migration(revisions_from=revisions_from)
+
+        instances = models.SamplePage.objects.all().annotate(
+            raw_content=Cast(F("content"), JSONField())
+        )
+
+        for instance in instances:
+            old_revisions = self.revisions[instance.id]
+            for old_revision, new_revision in zip(
+                old_revisions, instance.revisions.all().order_by("id")
+            ):
+                is_latest_or_live = (
+                    old_revision.id == instance.live_revision_id
+                    or old_revision.id == instance.latest_revision_id
+                )
+                is_after_revisions_from = (
+                    old_revision.created_at.timestamp() > revisions_from.timestamp()
+                )
+                is_altered = is_latest_or_live or is_after_revisions_from
+                old_content = json.loads(old_revision.content["content"])
+                new_content = json.loads(new_revision.content["content"])
+                for old_block, new_block in zip(old_content, new_content):
+                    self.assertEqual(old_block["id"], new_block["id"])
+                    if is_altered and old_block["type"] == "char1":
+                        self.assertEqual(new_block["type"], "renamed1")
+                    else:
+                        self.assertEqual(old_block["type"], new_block["type"])
+
+
+class TestNonPageModelWithRevisions(BaseMigrationTest):
+    model_name = "SampleModelWithRevisions"
+
+    @classmethod
+    def setUpTestData(cls):
+        instances = [None for i in range(3)]
+        instances[0] = factories.SampleModelWithRevisionsFactory(
+            content__0__char1="Test char 1",
+            content__1__char1="Test char 2",
+            content__2__char2="Test char 3",
+            content__3__char2="Test char 4",
+        )
+        instances[1] = factories.SampleModelWithRevisionsFactory(
+            content__0__char1="Test char 1",
+            content__1__char1="Test char 2",
+            content__2__char2="Test char 3",
+        )
+        instances[2] = factories.SampleModelWithRevisionsFactory(
+            content__0__char2="Test char 1",
+            content__1__char2="Test char 2",
+            content__2__char2="Test char 3",
+        )
+
+        cls.instances = {}
+        cls.revisions = {}
+        for instance in instances:
```

Comment:
Maybe break out revision creation to its own method or utility function, as you're writing this out a couple of times (unless there's something different going on each time)?